graphql-server/go-server/pkg/llm/llm.go

```diff
@@ -0,0 +1,149 @@
+	condition := llm.Status.GetCondition(v1alpha1.TypeReady)
+	updateTime := condition.LastTransitionTime.Time
+	status := string(condition.Status)
+	message := string(condition.Message)
+
+	llmType := string(llm.Spec.Type)
+
+	provider := string(llm.Spec.Provider.GetType())
+
+	md := generated.Llm{
+		ID:                &id,
+		Name:              obj.GetName(),
+		Namespace:         obj.GetNamespace(),
+		CreationTimestamp: &creationtimestamp,
+		Labels:            graphqlutils.MapStr2Any(obj.GetLabels()),
+		Annotations:       graphqlutils.MapStr2Any(obj.GetAnnotations()),
+		DisplayName:       &llm.Spec.DisplayName,
+		Description:       &llm.Spec.Description,
+		Type:              &llmType,
+		Status:            &status,
+		Message:           &message,
+		Provider:          &provider,
+		UpdateTimestamp:   &updateTime,
+	}
+	return &md
+}
+
+func ListLLMs(ctx context.Context, c dynamic.Interface, input generated.ListCommonInput) (*generated.PaginatedResult, error) {
+	keyword, labelSelector, fieldSelector := "", "", ""
+	page, pageSize := 1, 10
+	if input.Keyword != nil {
+		keyword = *input.Keyword
+	}
+	if input.FieldSelector != nil {
+		fieldSelector = *input.FieldSelector
+	}
+	if input.LabelSelector != nil {
+		labelSelector = *input.LabelSelector
+	}
+	if input.Page != nil && *input.Page > 0 {
+		page = *input.Page
+	}
+	if input.PageSize != nil && *input.PageSize > 0 {
+		pageSize = *input.PageSize
+	}
+
+	dsSchema := schema.GroupVersionResource{Group: v1alpha1.GroupVersion.Group, Version: v1alpha1.GroupVersion.Version, Resource: "llms"}
+	listOptions := metav1.ListOptions{
+		LabelSelector: labelSelector,
+		FieldSelector: fieldSelector,
+	}
+	us, err := c.Resource(dsSchema).Namespace(input.Namespace).List(ctx, listOptions)
+	if err != nil {
+		return nil, err
+	}
+	sort.Slice(us.Items, func(i, j int) bool {
+		return us.Items[i].GetCreationTimestamp().After(us.Items[j].GetCreationTimestamp().Time)
+	})
+
+	totalCount := len(us.Items)
+
+	result := make([]generated.PageNode, 0, pageSize)
+	pageStart := (page - 1) * pageSize
+
+	for index, u := range us.Items {
+		// skip if smaller than the start index
+		if index < pageStart {
+			continue
+		}
+		m := unstructured2LLM(&u)
+		// filter based on `keyword`
+		if keyword != "" {
+			if !strings.Contains(m.Name, keyword) && !strings.Contains(*m.DisplayName, keyword) {
+				continue
+			}
+		}
+		result = append(result, m)
+
+		// break if page size matches
+		if len(result) == pageSize {
+			break
+		}
+	}
+
+	end := page * pageSize
+	if end > totalCount {
+		end = totalCount
+	}
+
+	return &generated.PaginatedResult{
+		TotalCount:  totalCount,
+		HasNextPage: end < totalCount,
+		Nodes:       result,
+	}, nil
+}
+
+// ReadLLM
+func ReadLLM(ctx context.Context, c dynamic.Interface, name, namespace string) (*generated.Llm, error) {
+	resource := c.Resource(schema.GroupVersionResource{Group: v1alpha1.GroupVersion.Group, Version: v1alpha1.GroupVersion.Version, Resource: "llms"})
```

Comment:
Please use schemas in https://github.com/kubeagi/arcadia/blob/main/graphql-server/go-server/pkg/common/schema.go#L66